python_src/source/simulator/simulator.py

```diff
@@ -1,12 +1,48 @@
 import lgsvl
+import os
 
 
 class Simulator:
-    def __init__(self):
-        # Connect to the simulator. Load the scene from http://localhost:8080/#/Maps
-        sim = lgsvl.Simulator(address="localhost", port=8181)
-        if sim.current_scene == "SingleLaneRoad":
-            sim.reset()
-        else:
-            sim.load("SingleLaneRoad", seed=650387)
-        self.sim = sim
\ No newline at end of file
+    def __init__(self, seconds=30, scene="SingleLaneRoad", error_message=None, load_scene=True):
+        if error_message is None:
+            error_message = 'test timed out after {}s.'.format(seconds)
+        self.seconds = seconds
+        self.error_message = error_message
+        self.scene = scene
+        self.load_scene = load_scene
+
+    def __enter__(self):
+        self.sim = lgsvl.Simulator(os.environ.get("SIMULATOR_HOST", "127.0.0.1"), 8181)
+        if self.load_scene:
+            if self.sim.current_scene == self.scene:
+                self.sim.reset()
+            else:
+                self.sim.load(self.scene)
+        return self.sim
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.sim.close()
+
+
+def spawn_state(sim, index=0):
+    state = lgsvl.AgentState()
+    state.transform = sim.get_spawn()[index]
+    return state
+
+
+def create_EGO(sim, state=None):  # Only create an EGO is none are already spawned
+    if state is None:
+        state = spawn_state(sim)
+    return sim.add_agent("Jaguar2015XE (Apollo 3.0)", lgsvl.AgentType.EGO, state)
```

Comment:
Is there a special reason for using Apollo **3** instead of Apollo **5** or does the exact car not really matter? In case the vehicle config matters could you provide its vehicle config?

Reply:
Nice catch. I might rely on the old documentation. Will update it to 5 after a meeting 👍🏻

Reply:
I checked and realized `Jaguar2015XE (Apollo 3.0)` is the name of ego vehicle as we see in http://localhost:8080/#/Vehicles, thus it is compulsory to type an exact vehicle name in order to create the vehicle corresponding to the brand we want. In case we want to use Apollo 5, we can use the name `Lincoln2017MKZ (Apollo 5.0)`